ddtrace/settings/_otel_remapper.py

```diff
@@ -0,0 +1,184 @@
+
+        for key, value in otel_user_tag_dict.items():
+            if key.lower() not in OTEL_UNIFIED_TAG_MAPPINGS.keys():
+                if len(dd_tags) < 10:
+                    dd_tags.append("{}:{}".format(key, value))
+                else:
+                    remaining_tags.append("{}:{}".format(key, value))
+    except Exception:
+        log.warning("DDTRACE failed to read OTEL_RESOURCE_ATTRIBUTES. This value is misformatted: %s", otel_value)
+
+    if len(otel_user_tag_dict.items()) > 10:
+        log.warning(
+            "To preserve metrics cardinality, only the following first 10 tags have been processed %s. "
+            "The following tags were not ingested: %s",
+            dd_tags,
+            remaining_tags,
+        )
+    return ",".join(dd_tags)
+
+
+def _remap_otel_sdk_config(otel_value):
+    """Remaps the otel sdk config to ddtrace sdk config"""
+    if otel_value == "false":
+        return "True"
+    elif otel_value == "true":
+        return "False"
+    else:
+        log.warning("OTEL_SDK_DISABLED='%s'  is not supported", otel_value)
+        return otel_value
+
+
+def _remap_default(otel_value):
+    """Remaps the otel default value to ddtrace default value"""
+    return otel_value
+
+
+ENV_VAR_MAPPINGS = {
+    "OTEL_SERVICE_NAME": ("DD_SERVICE", _remap_default),
+    "OTEL_LOG_LEVEL": ("DD_TRACE_DEBUG", _remap_otel_log_level),
+    "OTEL_PROPAGATORS": ("DD_TRACE_PROPAGATION_STYLE", _remap_otel_propagators),
+    "OTEL_TRACES_SAMPLER": ("DD_TRACE_SAMPLE_RATE", _remap_traces_sampler),
+    "OTEL_TRACES_EXPORTER": ("DD_TRACE_ENABLED", _remap_traces_exporter),
+    "OTEL_METRICS_EXPORTER": ("DD_RUNTIME_METRICS_ENABLED", _remap_metrics_exporter),
+    "OTEL_LOGS_EXPORTER": ("", _remap_logs_exporter),  # Does not set a DDTRACE environment variable.
+    "OTEL_RESOURCE_ATTRIBUTES": ("DD_TAGS", _remap_otel_tags),
+    "OTEL_SDK_DISABLED": ("DD_TRACE_OTEL_ENABLED", _remap_otel_sdk_config),
+}
+
+
+def otel_remapping():
+    """Checks for the existence of both OTEL and Datadog tracer environment variables and remaps them accordingly.
+    Datadog Environment variables take precedence over OTEL, but if there isn't a Datadog value present,
+    then OTEL values take their place.
+    """
+    user_envs = {key.upper(): value for key, value in os.environ.items()}
+
+    for otel_env, otel_value in user_envs.items():
+        if otel_env not in ENV_VAR_MAPPINGS.keys():
+            continue
+
+        dd_env, otel_config_remapper = ENV_VAR_MAPPINGS[otel_env]
+        if dd_env in user_envs:
+            log.debug(
+                "Datadog configuration %s is already set. OpenTelemetry configuration will be ignored: %s=%s",
+                dd_env,
+                otel_env,
+                otel_value,
+            )
+            continue
+
+        if otel_env not in ["OTEL_RESOURCE_ATTRIBUTES", "OTEL_SERVICE_NAME"]:
+            # Resource attributes and service name are case-insensitive
```

Comment:
It's possible I'm misunderstanding the comment, but the comment and the if check look like they're doing opposite things to me. Can you confirm whether we should be setting `lower()` only when the key is not OTEL_RESOURCE_ATTRIBUTES or OTEL_SERVICE_NAME?

Reply:
@zacharycmontoya Yup! Can confirm this is the case; spoke with Munir as well and my thoughts were tags/service names should retain their case sensitivity for the customer's own consistency, while other vars would be "lowered" to for our own consistency